client/components/notes-formatted-block/test/index.js

```diff
@@ -1,72 +1,99 @@
-import { render, shallow } from 'enzyme';
+/**
+ * @jest-environment jsdom
+ */
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import FormattedBlock, { FormattedBlockRenderer } from '..';
 
 describe( 'FormattedBlock', () => {
 	beforeEach( () => jest.resetAllMocks() );
 
-	test( 'displays string content as-is', () => {
+	test( 'renders string content as-is', () => {
 		const content = 'my content';
-		const block = render( <FormattedBlock content={ content } /> );
 
-		expect( block.html() ).toEqual( content );
-	} );
+		render( <FormattedBlock content={ content } /> );
+
+		const block = screen.getByText( content );
 
-	test( 'displays content text as-is if there is no nested content and the content type is undefined', () => {
-		const text = 'my content text';
-		const block = render( <FormattedBlock content={ { text } } /> );
+		expect( block ).toBeInTheDocument();
 
-		expect( block.html() ).toEqual( text );
+		// Test it's rendered as a raw string
+		expect( block.innerHTML ).toEqual( content );
 	} );
 
-	test( 'displays nested content as FormattedBlocks if content type is not supported', () => {
-		const onClick = jest.fn();
-		const meta = {};
+	test( 'renders string content without using a custom block if content type is undefined', () => {
+		const MockBlockMapping = FormattedBlockRenderer( {
+			myBlock: ( props ) => (
+				<div type="myfakeblock" { ...props }>
+					MyFakeBlock
+				</div>
+			),
+		} );
+
 		const children = [
 			'example1',
 			{
-				text: 'example2',
+				text: 'main',
 			},
 			{
-				children: [ 'example3a', 'example3b' ],
+				children: [ 'children1', 'children2' ],
 			},
 		];
 
-		const block = shallow(
-			<FormattedBlock content={ { children } } onClick={ onClick } meta={ meta } />
-		);
-
-		expect( block.children().length ).toEqual( children.length );
-
-		expect(
-			block
-				.children()
-				.everyWhere(
-					( child ) =>
-						child.type() === FormattedBlock &&
-						child.prop( 'onClick' ) === onClick &&
-						child.prop( 'meta' ) === meta
-				)
-		).toEqual( true );
+		render( <MockBlockMapping content={ { children } } meta={ {} } /> );
+
+		const customBlock = screen.queryByText( 'MyFakeBlock' );
+		const block = screen.getByText( /example1/ );
+
+		expect( customBlock ).not.toBeInTheDocument();
+
+		expect( block ).toBeInTheDocument();
+		expect( block ).toHaveTextContent( 'main' );
+		expect( block ).toHaveTextContent( 'children1' );
+		expect( block ).toHaveTextContent( 'children2' );
 	} );
 
-	test( 'displays the correct block with correct props if the content type is supported', () => {
+	test( 'handles a click if `onClick` prop is passed', async () => {
+		userEvent.setup();
+
 		const MockBlockMapping = FormattedBlockRenderer( {
-			myBlock: ( props ) => <div type="myfakeblock" { ...props }></div>,
+			myBlock: ( props ) => (
+				<div type="myfakeblock" { ...props }>
```

Comment:
What's with the `type` attribute? Is it necessary? I think we can remove it everywhere. If necessary for querying, we can always use `data-testid` in the mock.

Reply:
Yep, you're right, it's not really used for anything! Not sure why it was there. I removed them.